FE/src/components/NewRecipe.jsx

```diff
@@ -1,8 +1,165 @@
+   function handleInstructionChange(e, index) {
+    e.preventDefault();
+    const {name, value} = e.target;
+    const list = [...instructionList];
+    list[index][name] = value;
+    setInstructionList(list);
+   };
+
+   function handleNoteChange(e, index) {
+    e.preventDefault();
+    const {name, value} = e.target;
+    const list = [...notesList];
+    list[index][name] = value;
+    setNotesList(list);
+   };
+
+   function rearrange (input, field) {
+    let output = [];
+    for (let i=0; i < input.length; i++) {
+        output.push(input[i][field]);
+    }
+    return output;
+   }
+
+   function handleSubmit(e) {
+        e.preventDefault();
+
+        let ingredArray = rearrange(ingredientList, 'ingredient');
+        let instructArray = rearrange(instructionList, 'instruction');
+        let noteArray = rearrange(notesList, 'note');
+
+
+        let data = {
+            title: title,
+            estTime: estTime,
+            ingredients: ingredArray,
+            procedure: instructArray,
+            image: image,
+            notes: noteArray,
+            tags: tags
+        };
+
+        console.log(data)
+    }
+
+   return (
+       <div>
+           <form onSubmit={handleSubmit}>
+               <label>Title: </label>
+               <input type="text" id="title" name="title" value={title} onChange={(e) => {
+                setTitle(e.target.value);
+               }}/>
+
+               <label>Estimated Time: </label>
+               <input list="times" id="estTime" name="estTime" value={estTime} onChange={(e) => {
+                setEstTime(e.target.value);
+               }}/>
+                   <datalist id="times">
+                       <option value="15 min"/>
+                       <option value="30 min"/>
+                       <option value="45 min"/>
+                       <option value="60 min"/>
+                       <option value="75 min"/>
+                       <option value="90 min"/>
+                   </datalist>
+
+               <label>Ingredients: </label>
+               {ingredientList.map((singleIngred, index) => {
+                return (
+                <div key={index}>
+                    <input type="text" name="ingredient"  value={singleIngred.ingredient} onChange={(e) => handleIngredientChange(e, index)}/>
+                    {ingredientList.length > 1 && <button type="button" onClick={() => handleIngredientDelete(index)}>-</button>}
+                    {ingredientList.length - 1 === index && ingredientList.length < 20 && <button type="button" onClick={handleIngredientAdd}>+</button>}
+                </div>
+                
+               )})}
+
+               <label>Instructions: </label>
+               {instructionList.map((singleInstruct, index) => {
+                return (
+                    <div key={index}>
+                        <input type="text" name="instruction" value={singleInstruct.instruction} onChange={(e) => handleInstructionChange(e, index)}/>
```

Comment:
It's a little unclear visually right now that they need to put in a single step then hit the + button. Maybe we should add  something like `Step ${num}:`